### PR TITLE
Make the code finally compile with tests

### PR DESCRIPTION
**Pull request by rafaelcn**

### CODE REVIEW

**Review by ryukinix — APPROVED** [review]
Overall looks good, I only requested some clarifications about good practices for c++. Feel free to respond now or later after merge.

File: kokiri/vendor/glad/src/glad.c [hunk 1]
```diff
@@ -838,7 +838,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "glad/glad.h"
+#include <glad/glad.h>
```
[the remark this answers is not in the document]
Comment (rafaelcn) [reply kokiri/vendor/glad/src/glad.c:841]: It doesn't matter that much, is just a preference style. Things that are not from Kokiri itself might be assumed to be from the system and in this case we vendor all of our dependencies, in the end it doesn't matter.

File: kokiri/src/graphics/window.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <string>
+#include <cinttypes>
+
+#include <SDL2/SDL.h>
```
[the remark this answers is not in the document]
Comment (rafaelcn) [reply kokiri/src/graphics/window.hpp:6]: This is one of the libraries not included in the vendor folder and it depends on how the system itself installs the SDL library, I do think that it is a good idea to encapsulate all of the SDL headers in a folder so...

File: kokiri/src/graphics/opengl/renderer2d.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#if defined(__WIN32__) || defined(__WIN32) || defined(__MINGW32__)
+#include <GL/glext.h>
+#endif
+
+
+#include "context.hpp"
+
+#include "../window.hpp"
```
[the remark this answers is not in the document]
Comment (rafaelcn) [reply kokiri/src/graphics/opengl/renderer2d.hpp:10]: I don't have any reason off the top of my head to not use it nor any reason why it is alleged to be bad. That being said, I don't think it's a bad idea.

**Comment by ryukinix** [comment]
If you say that, I am ok!

Cordialmente,
Manoel Vilela.

Em sáb., 1 de abr. de 2023 22:22, Rafael Campos Nunes <
***@***.***> escreveu:

**Pull request merged**